source/h2_decoder.c

```diff
@@ -1263,6 +1282,30 @@ static int s_process_header_field(struct aws_h2_decoder *decoder, const struct a
 
         /* #TODO Validate characters used in header_field->value */
 
+        if (name_enum == AWS_HTTP_HEADER_COOKIE) {
```

Comment:
trivial/efficiency: `switch(name_enum)`
we'll probably add special code for more headers in the future, and compilers can sometimes do cool tricks with switch statements